src/interpreter.ts

```diff
@@ -25,3 +499,840 @@ export function parseAndEvalExpression(sourceCode: string): EvalResult {
+        throwNonFatalErrorConstEval(
+            "Primitive type declarations are currently not supported.",
+            ast.loc,
+        );
+    }
+
+    public interpretFunctionDecl(ast: AstNativeFunctionDecl) {
+        throwNonFatalErrorConstEval(
+            "Native function declarations are currently not supported.",
+            ast.loc,
+        );
+    }
+
+    public interpretContract(ast: AstContract) {
+        throwNonFatalErrorConstEval(
+            "Contract declarations are currently not supported.",
+            ast.loc,
+        );
+    }
+
+    public interpretTrait(ast: AstTrait) {
+        throwNonFatalErrorConstEval(
+            "Trait declarations are currently not supported.",
+            ast.loc,
+        );
+    }
+
+    public interpretExpression(ast: AstExpression): Value {
+        switch (ast.kind) {
+            case "id":
+                return this.interpretName(ast);
+            case "method_call":
+                return this.interpretMethodCall(ast);
+            case "init_of":
+                return this.interpretInitOf(ast);
+            case "null":
+                return this.interpretNull(ast);
+            case "boolean":
+                return this.interpretBoolean(ast);
+            case "number":
+                return this.interpretNumber(ast);
+            case "string":
+                return this.interpretString(ast);
+            case "op_unary":
+                return this.interpretUnaryOp(ast);
+            case "op_binary":
+                return this.interpretBinaryOp(ast);
+            case "conditional":
+                return this.interpretConditional(ast);
+            case "struct_instance":
+                return this.interpretStructInstance(ast);
+            case "field_access":
+                return this.interpretFieldAccess(ast);
+            case "static_call":
+                return this.interpretStaticCall(ast);
+        }
+    }
+
+    public interpretName(ast: AstId): Value {
+        if (hasStaticConstant(this.initialContext, idText(ast))) {
+            const constant = getStaticConstant(
+                this.initialContext,
+                idText(ast),
+            );
+            if (constant.value !== undefined) {
+                return constant.value;
+            } else {
+                throwErrorConstEval(
+                    `cannot evaluate declared constant ${idTextErr(ast)} as it does not have a body`,
+                    ast.loc,
+                );
+            }
+        }
+        const letBinding = this.envStack.getBinding(idText(ast));
+        if (letBinding !== undefined) {
+            return letBinding;
+        }
+        throwNonFatalErrorConstEval("cannot evaluate a variable", ast.loc);
+    }
+
+    public interpretMethodCall(ast: AstMethodCall): Value {
+        const methodName = ast.method;
+        const args = ast.args;
+        const source = ast.loc;
+        const object = ast.self;
```

Comment:
I'd prefer not to have any local aliases in such cases, because you need to track renames when reading the source code. Also, `loc` is sort of standard, so I don't get the point of renaming it.

```suggestion
```